### PR TITLE
SNS - ADC Logging

### PR DESCRIPTION
**Pull request by kshxtij**
Add logging interface to `io::ADC` and replace all comments with actual logging

### CODE REVIEW

File: lib/io/adc.cpp [hunk 1]
```diff
@@ -5,45 +5,49 @@
...
 }
 
-std::optional<uint16_t> resetAndRead4(const int file_descriptor)
+std::optional<uint16_t> Adc::resetAndRead4(const int file_descriptor)
 {
-  const auto offset = lseek(file_descriptor, 0, SEEK_SET);        // reset file pointer
+  const auto offset = lseek(file_descriptor, 0, SEEK_SET);  // reset file pointer
   if (offset != 0) {
-    /*TODO: log that we failed in resetting file_descriptor's file offset*/
+    logger_.log(hyped::core::LogLevel::kInfo, "Failed to reset file offset");
```
Comment (SnickeyX) [inline lib/io/adc.cpp:37]: How are we differentiating between info and fatal errors? This seems just as bad as not being able to open the ADC file.

File: lib/io/adc.cpp [hunk 2]
```diff
@@ -5,45 +5,49 @@
...
-                                  sizeof(read_buffer));           // actually consume new data, changes value in buffer
+  char read_buffer[4];  // buffer size 4 for fs value
+  const int num_bytes_read
+    = read(file_descriptor,
+           &read_buffer,
+           sizeof(read_buffer));  // actually consume new data, changes value in buffer
   if (num_bytes_read != sizeof(read_buffer)) {
-    /*TODO: log that we failed in reading 4 bytes or reached EOF*/
-    return std::nullopt;                                          // returning NULL since we did not get any value
+    logger_.log(hyped::core::LogLevel::kInfo, "Failed to read 4 bytes");
```
Comment (SnickeyX) [inline lib/io/adc.cpp:46]: Same thing here

**Review by SnickeyX — CHANGES_REQUESTED** [review]

**Review by TomLonergan03 — COMMENTED** [review]
I agree with Ishan on levels, once that's done I'm happy with it

**Review by SnickeyX — APPROVED** [review]
looks good

File: lib/io/adc.cpp [hunk 3]
```diff
@@ -5,45 +5,49 @@
 
 namespace hyped::io {
 
-Adc::Adc(const uint8_t pin) : pin_(pin)
+Adc::Adc(const uint32_t pin, hyped::core::ILogger &logger) : pin_(pin), logger_(logger)
```
Comment (SnickeyX) [inline lib/io/adc.cpp:8]: uint8_t pls

File: lib/io/adc.hpp [hunk 4]
```diff
@@ -5,14 +5,16 @@
...
+
 namespace hyped::io {
 
 class Adc {
  public:
   /**
    * @param pin is one of the 6 analogue input pins on the bbb
    */
-  Adc(const uint8_t pin);
+  Adc(const uint32_t pin, hyped::core::ILogger &logger);
```
Comment (SnickeyX) [inline lib/io/adc.hpp:17]: same here

**Review by SnickeyX — APPROVED** [review]

**Review by TomLonergan03 — APPROVED** [review]

**Pull request merged**